packages/aws-library/src/aws_library/s3/client.py

```diff
@@ -53,21 +55,19 @@ async def http_check_bucket_connected(self, bucket: S3BucketName) -> bool:
         except Exception:  # pylint: disable=broad-except
             return False
 
-    async def create_presigned_download_link(
+    @s3_exception_handler(_logger)
+    async def create_single_presigned_download_link(
         self,
         bucket_name: S3BucketName,
         object_key: str,
         expiration_secs: int,
     ) -> AnyUrl:
-        try:
-            # NOTE: ensure the bucket/object exists, this will raise if not
-            await self.client.head_bucket(Bucket=bucket_name)
-            generated_link = await self.client.generate_presigned_url(
-                "get_object",
-                Params={"Bucket": bucket_name, "Key": object_key},
-                ExpiresIn=expiration_secs,
-            )
-            url: AnyUrl = parse_obj_as(AnyUrl, generated_link)
-            return url
-        except botocore.exceptions.ClientError as exc:
-            raise S3RuntimeError from exc  # pragma: no cover
```

Comment:
you are replacing this error with S3AccessError. I would not do so here!
Please make sure the interface does not change

Reply:
Yes that is the reason for this change, so I keep the same interface as it is now in the Storage :+1: